presto-verifier/src/test/java/com/facebook/presto/verifier/checksum/TestChecksumValidator.java

```diff
@@ -328,4 +337,53 @@ public void testArray()
                         .put(MAP_ARRAY_COLUMN, new ColumnMatchResult(false, "control(checksum: 0b, cardinality_sum: 7) test(checksum: 1b, cardinality_sum: 5)"))
                         .build());
     }
+
+    @Test
+    public void testRow()
+    {
+        List<Column> columns = ImmutableList.of(ROW_COLUMN);
+        ChecksumResult controlChecksum = new ChecksumResult(
+                5,
+                ImmutableMap.<String, Object>builder()
+                        .put("row_checksum", new SqlVarbinary(new byte[] {0xa}))
+                        .put("row$a_checksum", new SqlVarbinary(new byte[] {0xb}))
+                        .put("row$_col2_checksum", new SqlVarbinary(new byte[] {0xc}))
+                        .put("row$_col3_checksum", new SqlVarbinary(new byte[] {0xd}))
+                        .put("row$b_checksum", new SqlVarbinary(new byte[] {0xe}))
+                        .build());
+
+        assertTrue(checksumValidator.getMismatchedColumns(columns, controlChecksum, controlChecksum).isEmpty());
+
+        // Mismatched different elements
+        ChecksumResult testChecksum = new ChecksumResult(
+                5,
+                ImmutableMap.<String, Object>builder()
+                        .put("row_checksum", new SqlVarbinary(new byte[] {0x1a}))
+                        .put("row$a_checksum", new SqlVarbinary(new byte[] {0x1b}))
+                        .put("row$_col2_checksum", new SqlVarbinary(new byte[] {0x1c}))
+                        .put("row$_col3_checksum", new SqlVarbinary(new byte[] {0x1d}))
```

Comment:
Sorry about my previous suggestion for the column name but this just looks a bit confusing. Maybe we should do `$col2` instead of `_col2`, so that this looks like `row$$col3_checksum`, and for the named field, it should still be `row$a_checksum`. The reason we add an extra special character before the field name is that this will be safe even if a column in the row was explicitly named as `col1`, `col2`, etc...